### PR TITLE
feat: Cudos-798 CosmWasm module implementation

### PR DESCRIPTION
**Pull request by avalkov**

### CODE REVIEW

File: database/cosmwasm.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package database
+
+import "github.com/forbole/bdjuno/v2/types"
+
+func (db *Db) SaveMsgStoreCodeData(data *types.MsgStoreCodeData) error {
```
Comment (Infra1515) [inline database/cosmwasm.go:5]: Possibly think of how to fetch all the cosmwasm tx with one call. If you can do it with GQL by tweaking the query, this is fine. If not - think about creating a base table that holds all the CW txs

File: database/scheme/13-cosmwasm.sql [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+CREATE INDEX cosmwasm_instantiate_sender_index ON cosmwasm_instantiate (sender);
+CREATE INDEX cosmwasm_instantiate_code_id_index ON cosmwasm_instantiate (code_id);
+CREATE INDEX cosmwasm_instantiate_result_contract_address_index ON cosmwasm_instantiate (result_contract_address);
+
+CREATE TABLE cosmwasm_execute
+(
+    transaction_hash TEXT NOT NULL,
+    index INTEGER NOT NULL,
+    method TEXT NOT NULL,
+    arguments JSONB,
```
Comment (Infra1515) [inline database/scheme/13-cosmwasm.sql]: Possibly validate all the incoming JSONB types and sanitize them as needed like so
func sanitize(s string) string {
    return strings.ToValidUTF8(s, "")
}

File: database/scheme/13-cosmwasm.sql [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+CREATE INDEX cosmwasm_execute_contract_index ON cosmwasm_execute (contract);
+
+CREATE TABLE cosmwasm_migrate
+(
+    transaction_hash TEXT NOT NULL,
+    index INTEGER NOT NULL,
+    sender TEXT NOT NULL,
+    contract TEXT NOT NULL,
+    code_id TEXT NOT NULL,
+    arguments JSONB,
```
Comment (Infra1515) [inline database/scheme/13-cosmwasm.sql]: same as above

File: modules/cosmwasm/handle_msg.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+			msg.Contract,
+		),
+	)
+}
+
+func isSuccess(code uint32) bool {
+	return code == 0
+}
+
+func getValueFromLogs(index uint32, logs sdk.ABCIMessageLogs, eventType, attributeKey string) string {
```
Comment (Infra1515) [inline modules/cosmwasm/handle_msg.go:175]: add unit tests for this method

File: modules/cosmwasm/handle_msg.go [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+					return attr.Value
+				}
+			}
+		}
+	}
+
+	return ""
+}
+
+func getPayloadMapKeys(payloadMap map[string]interface{}) []string {
```
Comment (Infra1515) [inline modules/cosmwasm/handle_msg.go:197]: unit tests as well

File: modules/gov/handle_genesis.go [hunk 6]
```diff
@@ -24,7 +25,7 @@ func (m *Module) HandleGenesis(doc *tmtypes.GenesisDoc, appState map[string]json
 	}
 
 	// Save the proposals
-	err = m.saveProposals(genState.Proposals)
+	err = m.saveProposals(genState.Proposals, doc.InitialHeight)
```
Comment (Infra1515) [inline modules/gov/handle_genesis.go:28]: This is reasonable, possibly add a PR to the original repo with this change

File: cmd/parse-genesis/parseGenesis.go [hunk 7]
```diff
@@ -45,6 +47,10 @@ func NewParseGenesisCmd(parseCfg *parse.Config) *cobra.Command {
 				return fmt.Errorf("error while getting genesis doc and state: %s", err)
 			}
 
+			if err := resolveInitialHeightBlock(parseCtx, genesisDoc.InitialHeight); err != nil {
```
Comment (Infra1515) [inline cmd/parse-genesis/parseGenesis.go:51]: check if block already exists, it it does - skip this function

File: cmd/parse-genesis/parseGenesis.go [hunk 8]
```diff
@@ -69,3 +75,53 @@ func getModule(module string, parseCtx *parse.Context) (modules.Module, bool) {
...
+	if err != nil {
+		return fmt.Errorf("failed to get transactions for block: %s", err)
+	}
+
+	vals, err := ctx.Node.Validators(initialHeight)
+	if err != nil {
+		return fmt.Errorf("failed to get validators for block: %s", err)
+	}
+
+	isValidatorFound := false
```
Comment (Infra1515) [inline cmd/parse-genesis/parseGenesis.go:117]: move this for cycle to a function so it is more readable

**Review by Infra1515 — CHANGES_REQUESTED** [review]
Looks good, just check the left remarks

File: cmd/parse-genesis/parseGenesis.go (hunk 7 above)
Comment (avalkov) [reply cmd/parse-genesis/parseGenesis.go:51]: Fixed

File: modules/cosmwasm/handle_msg.go (hunk 5 above)
Comment (avalkov) [reply modules/cosmwasm/handle_msg.go:197]: Fixed

File: modules/cosmwasm/handle_msg.go (hunk 4 above)
Comment (avalkov) [reply modules/cosmwasm/handle_msg.go:175]: Fixed

File: database/scheme/13-cosmwasm.sql (hunk 3 above)
Comment (avalkov) [reply database/scheme/13-cosmwasm.sql]: Fixed

File: database/scheme/13-cosmwasm.sql (hunk 2 above)
Comment (avalkov) [reply database/scheme/13-cosmwasm.sql]: Fixed

File: database/cosmwasm.go (hunk 1 above)
Comment (avalkov) [reply database/cosmwasm.go:5]: As we discussed, we will do additional optimizations if required when we start to implement the UI.

**Pull request merged**